packages/analytics/analytics-chart/src/components/TopNTable.vue

```diff
@@ -0,0 +1,268 @@
+<template>
+  <KCard class="kong-ui-public-top-n-table">
+    <template
+      v-if="title"
+      #title
+    >
+      <div data-testid="top-n-card-title">
+        {{ title }}
+      </div>
+    </template>
+    <template
+      v-if="description"
+      #actions
+    >
+      <div
+        class="top-n-card-description"
+        data-testid="top-n-card-description"
+      >
+        {{ description }}
+      </div>
+    </template>
+
+    <template #body>
+      <KEmptyState
+        v-if="errorMessage"
+        cta-is-hidden
+        data-testid="top-n-error-state"
+        is-error
+      >
+        <template #title>
+          {{ i18n.t('topNTable.defaultErrorStateTitle') }}
+        </template>
+        <template #message>
+          {{ errorMessage }}
+        </template>
+      </KEmptyState>
+
+      <KEmptyState
+        v-else-if="!hasData"
+        cta-is-hidden
+        data-testid="top-n-empty-state"
+        icon="stateNoData"
+        icon-size="80"
+      >
+        <template #title>
+          {{ emptyStateTitle || i18n.t('topNTable.defaultEmptyStateTitle') }}
+        </template>
+      </KEmptyState>
+
+      <div
+        v-else
+        class="top-n-table"
+        data-testid="top-n-table"
+      >
+        <div class="table-headings">
+          <div class="table-row">
+            <div class="column-1">
+              {{ i18n.t('topNTable.nameLabel') }}
+            </div>
+            <div class="column-2">
+              {{ columnName }}
+            </div>
+          </div>
+        </div>
+
+        <div
+          class="table-body"
+          data-testid="top-n-data"
+        >
+          <div
+            v-for="(entry, idx) in records"
+            :key="`entry-${idx}`"
+            class="table-row"
+            :data-testid="`row-${getId(entry)}`"
+          >
+            <div class="column-1">
+              <slot
+                name="name"
+                :record="{
+                  id: getId(entry),
+                  name: getName(entry),
+                  dimension: displayKey,
+                }"
+              >
+                {{ getName(entry) }}
+              </slot>
+            </div>
+            <div class="column-2">
+              &nbsp; {{ getValue(entry) }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </template>
+  </KCard>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import { AnalyticsExploreV3Result, AnalyticsExploreV3Meta, AnalyticsExploreRecord, RecordEvent } from '@kong-ui-public/analytics-utilities'
```

Comment:
fix: import type (this will soon be enforced via https://github.com/Kong/public-ui-components/pull/729)
```suggestion
import type { AnalyticsExploreV3Result, AnalyticsExploreV3Meta, AnalyticsExploreRecord, RecordEvent } from '@kong-ui-public/analytics-utilities'
```